src/Analysis/Ast/Impl/Analyzer/Expressions/ExpressionFinder.cs

```diff
@@ -189,6 +189,48 @@ private class NormalExpressionWalker : ExpressionWalker {
                 return false;
             }
 
+            public override bool Walk(ListComprehension node) {
+                if (!base.Walk(node)) {
+                    return false;
+                }
+
+                // Don't set scope if 2x 
+                if (!Tree.LanguageVersion.Is3x()) {
+                    return true;
+                }
+
+                Scope = node;
+                return true;
+            }
+
+            public override bool Walk(DictionaryComprehension node) {
+                if (!base.Walk(node)) {
+                    return false;
+                }
+
+                // Don't set scope if 2x 
+                if (!Tree.LanguageVersion.Is3x()) {
+                    return true;
+                }
+
+                Scope = node;
+                return true;
+            }
+
+            public override bool Walk(SetComprehension node) {
+                if (!base.Walk(node)) {
+                    return false;
+                }
+
+                // Don't set scope if 2x 
+                if (!Tree.LanguageVersion.Is3x()) {
+                    return true;
+                }
+
+                Scope = node;
+                return true;
+            }
```

Comment:
All three above is the same code, perhaps make function with `Comprehension` base parameter?

Reply:
Yeah I was wondering - I would have to edit the interface in PythonWalker to do that. Otherwise I guess I could make a common function and have the overrides redirect to it.

Reply:
Do you? All of the comprehension types have a common base, `Comprehension`. Generators, no (but I don't see that you've touched generators either).

Reply:
Ok I'll do the redirect way